src/mlagility/api/cloud.py

```diff
@@ -365,6 +396,82 @@ def execute_gpu_remotely(
     sys.stdout = sys.stdout.terminal
 
 
+def execute_gpu_locally(
+    state: build.State, log_execute_path: str, iterations: int
+) -> None:
+    """
+    Execute Model on the local GPU
+    """
+
+    # Redirect all stdout to log_file
+    sys.stdout = build.Logger(log_execute_path)
+
+    # Setup local execution folders to save outputs/ errors
+    username = getpass.getuser()
+    output_dir = f"/home/{username}/mlagility_local_cache"
```

Comment:
Is the `getpass` library necessary? Or could you just use `$USER`?

How does `getpass` work with the github test agent?